### DISCUSSION TITLE
靜態網站產生器方案的選擇 -> Docusaurus

### DISCUSSION THREAD

**Discussion by gasolin** [lun-dao/LunDAO#15] [General]
作為一個目的為產出中深度文章的DAO, 選擇靜態網站產生器也是個重點?

我建議從支援以下兩者為基礎來挑選

1. 可以 host 在 github 上

用類似 EIP 的形式，開 PR 投稿，
直接 host 在 https://github.com/lun-dao/LunDAO gh-pages branch 
這樣產出更清楚

2. 可以自動透過 github action 產生網站

文章被接受 (merge) 後自動化發布，減少手動工作

第三個考量是是否要規定為 Markdown 格式

https://jamstack.org/generators/ 上列出了一些選擇

常見的

- Gatsby https://jamstack.org/generators/gatsby/
- Docusaurus https://docusaurus.io/

**Comment by gasolin** [comment]
個人推薦用 Docusaurus，可以拿來建首頁 + Blog（文章） + Doc（電子書）， https://docusaurus.io/docs 說明完整，支援 plugin

而且也已有實際透過 github action 自動 deploy 的經驗 https://blog.gasolin.idv.tw/tools/blog-to-docusaurus

**Reply by yurenju** [reply]
看到你從 hexo 搬到 Docusaurus，你覺得比較起來如何呀？

**Reply by gasolin** [reply]
Hexo 專注 blog, 加了一堆 plugin 後, 升級時要一併修改 template 很痛苦.
Blog 有支援站內搜尋 / tag cloud

Docusaurus 沒 Hexo 那麼多變化, 但預設模板就不錯看了, 換個佈景顏色主題即可
沒有好的站內搜尋 / tag cloud (或只是我沒找到)

**Reply by yurenju** [reply]
如果會產生 static files 感覺搜尋讓 google 來做就好了也行。

**Comment by yurenju** [comment]
確實我們也需要想一下靜態產生器要用哪個，對我來說最重要的事情是要支援 Markdown，但這個大部分靜態產生器應該都可以，所以我就比較沒有偏好，如果可以支援 RSS 也不錯，但不是必要條件。

**Reply by gasolin** [reply]
靜態產生器幾乎都有支援 RSS

**Reply by gasolin** [reply]
RSS Feed https://lundao.tech//blog/rss.xml

**Comment by yurenju** [comment]
@gasolin 想請你發個 PR 登記一下 account address https://github.com/lun-dao/LunDAO/blob/main/contributors.csv

**Reply by gasolin** [reply]
https://github.com/lun-dao/LunDAO/pull/17
